### PR TITLE
refactor commands_test to choose a free port for the service protocol

### PR DESCRIPTION
**Pull request by devoncarew**
Have `bin/tasks/commands_test.dart` select an open port to use for the service protocol. I think this will make the devicelab test more reliable, and may be the reason why https://github.com/flutter/flutter/pull/10966 had devicelab errors.

### CODE REVIEW

File: dev/devicelab/bin/tasks/commands_test.dart [hunk 1]
```diff
@@ -13,10 +13,10 @@ import 'package:flutter_devicelab/framework/adb.dart';
 import 'package:flutter_devicelab/framework/framework.dart';
 import 'package:flutter_devicelab/framework/utils.dart';
 
-const int kObservatoryPort = 8888;
-
 void main() {
   task(() async {
+    final int vmServicePort = await findAvailablePort();
```
Comment (Hixie) [inline dev/devicelab/bin/tasks/commands_test.dart]: This isn't a valid solution to this problem. By the time the port is re-opened by the process that we want to connect to, the port may have been opened by something else.

What we should do instead is have the target app open a port of its own choice and then connect to that based on what it reports in the console.

Comment (devoncarew) [inline dev/devicelab/bin/tasks/commands_test.dart]: Cool; I updated this to let the app bind to a free port, and discover the port from the text from the launched app.

**Comment by Hixie** [comment]
LGTM

If there's other tests in that directory using "--observatory-port", please file a bug or add a TODO or something pointing to this solution so that we remember to fix those too (or fix them, if you're feeling particularly in need of yak hair; @yjbanov's already suggested that we need to factor out a bunch of this code for easy re-use anyway).

**Comment by devoncarew** [comment]
Sure - just one other place in `routing_test.dart`. I added a todo:. I'll want to work on landing a test for frame events, so will probably refactor common code at that time.

**Pull request merged**